ilastik/applets/dataSelection/datasetDetailedInfoTableView.py

```diff
@@ -167,20 +176,77 @@ def paint(self, painter, option, index):
                 button.addPrecomputedVolumeRequested.connect(
                     partial(parent_view.handleCellAddPrecomputedVolumeEvent, button)
                 )
-                button.addRemoteVolumeRequested.connect(partial(parent_view.handleCellAddRemoteVolumeEvent, button))
+                button.addDvidVolumeRequested.connect(partial(parent_view.handleCellAddDvidVolumeEvent, button))
                 parent_view.setIndexWidget(index, button)
-        elif index.data() != "":
-            if button is not None:
-                # If this row has data, we must delete the button.
-                # Otherwise, it can steal input events (e.g. mouse clicks) from the cell, even if it is hidden!
-                # However, we can't remove it yet, because we are currently running in the context of a signal handler for the button itself!
-                # Instead, use a QTimer to delete the button as soon as the eventloop is finished with the current event.
-                QTimer.singleShot(750, lambda: parent_view.setIndexWidget(index, None))
-        super(AddButtonDelegate, self).paint(painter, option, index)
+        elif index.data() != "" and button is not None:
+            # If this row has data, we must delete the button.
+            # Otherwise, it can steal input events (e.g. mouse clicks) from the cell, even if it is hidden!
+            # However, we can't remove it yet, because we are currently running in the context of a signal handler for the button itself!
+            # Instead, use a QTimer to delete the button as soon as the eventloop is finished with the current event.
+            QTimer.singleShot(750, lambda: parent_view.setIndexWidget(index, None))
+        super().paint(painter, option, index)
+
+
+class ScaleComboBoxDelegate(QStyledItemDelegate):
+    def paint(self, painter, option, index):
+        scales = index.model().get_scale_options(index.row())
+        if scales:
+            self.parent().openPersistentEditor(index)
+        else:
+            self.parent().closePersistentEditor(index)
+            super().paint(painter, option, index)
+
+    def createEditor(self, parent: "DatasetDetailedInfoTableView", option, index):
+        model: "DatasetDetailedInfoTableModel" = index.model()
+        scales = model.get_scale_options(index.row())
+        if not scales:
+            return None
+        combo = QComboBox(parent)
+        for scale_index, scale in enumerate(scales):
+            combo.addItem(scale, scale_index)
+        combo.currentIndexChanged.connect(partial(self.on_combo_selected, index))
+        return combo
+
+    def setEditorData(self, editor, index):
+        value = index.data(Qt.DisplayRole)
+        current_selected = editor.findText(value)
+        if current_selected >= 0:
+            editor.blockSignals(True)  # To avoid triggering on_combo_selected
```

Comment:
there is the `ilastik.utility.gui.widgets.silent_qobject` context manager that could be used here and in `on_combo_selected`  as well.